### PR TITLE
Yugg Gun to fix issues with gene sting

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Fix issues with using a spell for gene sting instead of gun delivering spell.

#### Describe the solution
reimplements my original work

### CODE REVIEW

File: data/json/monster_special_attacks/monster_gun.json [hunk 1]
```diff
@@ -99,6 +99,41 @@
...
+    "id": "gene_sting",
+    "type": "GUN",
+    "symbol": "%",
+    "color": "red",
+    "name": { "str": "barb launching organ" },
+    "description": "A mutated organ capable of launching bony barbs at great speed.",
+    "material": [ "hflesh" ],
+    "flags": [
+      "PRIMITIVE_RANGED_WEAPON",
+      "GENE_STING"
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/monster_gun.json:112]:
```suggestion
      "GENE_STING",
```

File: tools/json_tools/generic_guns_validator.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
     'chemical_spray',
     'fishspear',
     'flammable',
+    'gene_sting_gun',
```
Comment (Maleclypse) [inline tools/json_tools/generic_guns_validator.py:28]:
```suggestion
    'gene_sting_barb',
```

File: data/json/species.json [hunk 3]
```diff
@@ -60,6 +60,13 @@
     "description": "a nether creature",
     "flags": [ "DRACULIN_IMMUNE" ]
   },
+  {
+    "type": "SPECIES",
+    "id": "NETHER_BURROWING",
+    "description": "a nether creature",
+    "footsteps": "slithering",
```
Comment (Karol1223) [inline data/json/species.json:67]: I wouldn't go with slithering. I guess it'd be rumbling or rustling or whatever for burrowing, but slithering isn't the noise I'd expect to hear.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/species.json (hunk 3 above)
Comment (Maleclypse) [reply data/json/species.json:67]: Thank you! I couldn't think of any words other than "vibrations" and I was like that's gonna just be weird.

File: data/json/monster_special_attacks/monster_gun.json [hunk 4]
```diff
@@ -99,6 +99,40 @@
...
+    "weight": "540 g",
+    "volume": "750 ml",
+    "longest_side": "25 cm",
+    "to_hit": 1,
+    "reload_noise_volume": 2,
+    "loudness": 2,
+    "range": 12,
+    "dispersion": 100,
+    "durability": 8,
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "gene_sting_barb": 15 } } ],
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/monster_gun.json:133]:
```suggestion
    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "gene_sting": 15 } } ],
```

File: data/json/species.json (hunk 3 above)
Comment (Maleclypse) [inline data/json/species.json:67]:
```suggestion
    "footsteps": "rustling",
```

File: data/json/monster_special_attacks/monster_gun.json [hunk 5]
```diff
@@ -99,6 +99,40 @@
...
+      "NO_UNLOAD",
+      "RELOAD_AND_SHOOT"
+    ],
+    "skill": "pistol",
+    "ammo": [ "gene_sting" ],
+    "clip_size": 1,
+    "weight": "540 g",
+    "volume": "750 ml",
+    "longest_side": "25 cm",
+    "to_hit": 1,
```
Comment (Maleclypse) [inline data/json/monster_special_attacks/monster_gun.json:127]:
```suggestion
```

File: data/json/items/ammo/barb.json [hunk 6]
```diff
@@ -18,5 +18,25 @@
...
+    "material": [ "bone" ],
+    "volume": "250 ml",
+    "weight": "37 g",
+    "ammo_type": "gene_sting",
+    "damage": { "damage_type": "genetic_damage_ballistic", "amount": 5, "armor_penetration": 5 },
+    "dispersion": 120,
+    "loudness": 0,
+    "count": 10,
+    "show_stats": true,
+    "effects": [ "GENE_STING" ],
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:39]:
```suggestion
    "effects": [ "GENE_STING" ],
    "grip": "bad",
    "length": "hand",
    "surface": "pierce",
    "balance": "good"
```

File: data/json/items/ammo/barb.json [hunk 7]
```diff
@@ -18,5 +18,29 @@
...
+    "grip": "bad",
+    "length": "hand",
+    "surface": "pierce",
+    "balance": "good"
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:43]:
```suggestion
    "balance": "good",
```

File: data/json/items/ammo/barb.json [hunk 8]
```diff
@@ -18,5 +18,29 @@
...
+    "volume": "250 ml",
+    "weight": "37 g",
+    "ammo_type": "gene_sting",
+    "damage": { "damage_type": "genetic_damage_ballistic", "amount": 5, "armor_penetration": 5 },
+    "dispersion": 120,
+    "loudness": 0,
+    "count": 10,
+    "show_stats": true,
+    "effects": [ "GENE_STING" ],
+    "grip": "bad",
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:40]:
```suggestion
    "grip": "none",
```

File: data/json/items/ammo/barb.json [hunk 9]
```diff
@@ -18,5 +18,29 @@
...
+    "length": "hand",
+    "surface": "pierce",
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:42]:
```suggestion
```

File: data/json/items/ammo/barb.json [hunk 10]
```diff
@@ -18,5 +18,26 @@
...
+    "volume": "250 ml",
+    "weight": "37 g",
+    "ammo_type": "gene_sting",
+    "damage": { "damage_type": "genetic_damage_ballistic", "amount": 5, "armor_penetration": 5 },
+    "dispersion": 120,
+    "loudness": 0,
+    "count": 10,
+    "show_stats": true,
+    "effects": [ "GENE_STING" ],
+    "balance": "good",
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:40]:
```suggestion
```

File: data/json/items/ammo/barb.json [hunk 11]
```diff
@@ -18,5 +18,25 @@
...
+    "type": "AMMO",
+    "id": "gene_sting_barb",
+    "price": "4 USD",
+    "name": { "str": "gene sting barb" },
+    "symbol": "=",
+    "color": "cyan",
+    "description": "Tiny droplets of cyan venom trickle out of the sponge-like tissue of this sharp barb.",
+    "material": [ "bone" ],
+    "volume": "250 ml",
+    "weight": "37 g",
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:32]:
```suggestion
    "weight": "30 g",
```

File: data/json/items/ammo/barb.json [hunk 12]
```diff
@@ -18,5 +18,25 @@
...
+    "material": [ "bone" ],
+    "volume": "250 ml",
+    "weight": "30 g",
+    "ammo_type": "gene_sting",
+    "damage": { "damage_type": "genetic_damage_ballistic", "amount": 5, "armor_penetration": 5 },
+    "dispersion": 120,
+    "loudness": 0,
+    "count": 10,
+    "show_stats": true,
+    "effects": [ "GENE_STING" ],
```
Comment (Maleclypse) [inline data/json/items/ammo/barb.json:39]:
```suggestion
```

File: data/json/ammo_effects.json [hunk 13]
```diff
@@ -368,6 +368,11 @@
     "type": "ammo_effect",
     "explosion": { "power": 7980, "shrapnel": { "casing_mass": 8725, "fragment_mass": 0.5 } }
   },
+  {
+    "id": "GENE_STING",
+    "type": "ammo_effect",
+    "spell_data": { "id": "spell_gene_sting" }
+  },
```
Comment (Maleclypse) [inline data/json/ammo_effects.json:375]:
```suggestion
```

**Pull request merged**